src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,322 @@
+package de.tum.in.www1.artemis.web.rest.tutorialgroups;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.validation.Valid;
+import javax.ws.rs.BadRequestException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.config.Constants;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroup;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.tutorialgroups.TutorialGroupRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.TutorialGroupService;
+import de.tum.in.www1.artemis.service.dto.StudentDTO;
+import de.tum.in.www1.artemis.service.feature.Feature;
+import de.tum.in.www1.artemis.service.feature.FeatureToggle;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+@RestController
+@RequestMapping("/api")
+public class TutorialGroupResource {
+
+    private static final String ENTITY_NAME = "tutorialGroup";
+
+    private final Logger log = LoggerFactory.getLogger(TutorialGroupResource.class);
+
+    private final TutorialGroupService tutorialGroupService;
+
+    private final TutorialGroupRepository tutorialGroupRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final UserRepository userRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public TutorialGroupResource(AuthorizationCheckService authorizationCheckService, UserRepository userRepository, CourseRepository courseRepository,
+            TutorialGroupService tutorialGroupService, TutorialGroupRepository tutorialGroupRepository) {
+        this.tutorialGroupService = tutorialGroupService;
+        this.courseRepository = courseRepository;
+        this.userRepository = userRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.tutorialGroupRepository = tutorialGroupRepository;
+    }
+
+    /**
+     * GET /tutorial-groups/:tutorialGroupId/title : Returns the title of the tutorial-group with the given id
+     * <p>
+     * NOTE: Used by entity-title service in the client to resolve the title of a tutorial group for breadcrumbs
+     *
+     * @param tutorialGroupId the id of the tutorial group
+     * @return ResponseEntity with status 200 (OK) and with body containing the title of the tutorial group
+     */
+    @GetMapping("/tutorial-groups/{tutorialGroupId}/title")
```

Comment:
Unlike all other endpoints, this one does not include the `/courses/{courseId}` part. For consistency, you should consider adding it.

Reply:
@Hialus this does not work as the service resolving the breadcrumbs expects this structure of the endpoint.

Reply:
```
    private fetchTitle(type: EntityType, ids: number[]): void {
        let resourceUrl = 'api/';
        switch (type) {
            case EntityType.COURSE:
                resourceUrl += 'courses';
                break;
            case EntityType.EXERCISE:
                resourceUrl += 'exercises';
                break;
            case EntityType.LECTURE:
                resourceUrl += 'lectures';
                break;
            case EntityType.HINT:
                resourceUrl += `programming-exercises/${ids[1]}/exercise-hints`;
                break;
            case EntityType.DIAGRAM:
                resourceUrl += 'apollon-diagrams';
                break;
            case EntityType.EXAM:
                resourceUrl += 'exams';
                break;
            case EntityType.ORGANIZATION:
                resourceUrl += 'organizations';
                break;
        }

        this.http.get(`${SERVER_API_URL}${resourceUrl}/${ids[0]}/title`, { observe: 'response', responseType: 'text' }).subscribe((response: HttpResponse<string>) => {
            if (response.body) {
                this.setTitle(type, ids, response.body);
            }
        });
    }
```
In entity-title.service.ts

Reply:
From what I can see, this would not prevent using the path `courses/{id}/tutorial-groups/{id}`, as the exercise hints also use such a format.
But I am not that good regarding the client code, so I might be from here.